content/docs/guides/ha_scale/perf-scale.md

```diff
@@ -0,0 +1,107 @@
+---
+title: "Performance and Scalability"
+description: "Performance benchmark on control plane and data plane sidecar and scalability test of Open Service Mesh"
+type: docs
+---
+
+Open Service Mesh aims to create a high-performance service mesh while providing all the essential service mesh features. This document describes the tests performed to evaluate the performance and scalability of OSM.
+
+## Testing environment
+
+This test is performed on a Kubernetes cluster provisioned and managed by [Microsoft Azure Kubernetes Service](https://azure.microsoft.com/en-us/services/kubernetes-service/). The cluster has 100 Standard_DS2_v2 nodes (each node has 2 vCPUs and 7GB memory). It is ensured that there is only one OSM controller pod running with default resource quota during the test. OSM is configured in non-permissive traffic mode.
+
+## Process
+
+The testing mesh contains two types of services. One is a load generator (modified from [wrk2](https://github.com/giltene/wrk2)), which can send requests at specified rate, load balanced among all other services in the mesh. Currently we test HTTP 1 requests only. The load generator serves as the client side of a request. Throughout our test, there will be at most one load generator service. The other type of service is a simple echo application that replies with whatever it receives from an inbound request. It serves as the server side of a request. We can deploy multiple copies of the echo application in the mesh. In short, the mesh topology is a typical 1-to-N service architecture.
+
+<p align="center">
+  <img src="/docs/images/perf/mesh-topo.png" width="650"/>
+</p>
+
+
+## Performance
+
+In this test, we focus on the latency overhead and resource consumption added by the OSM sidecar. We currently test the per pod RPS at 100, 500, 1000, 2000 and 4000 respectively.
+
+### Latency
+
+Below is the request latency at different RPS levels with OSM installed. The added latency comes from both the client and server sidecars.
```

Comment:
There is one for latency without OSM but we need more discussion on it. It is a bit surprising that latency without OSM is higher in some cases. I believe this is because Envoy helps to reuse connections and saves time. Thus it's not a fair to directly comparing latency w/ and w/o OSM, and OSM does not support customizing data plane # of connections now.

Reply:
Thanks for explaining!